galley/pkg/config/analysis/analyzers/namespaceconflict/namespaceselector.go

```diff
@@ -0,0 +1,202 @@
+type Analyzer struct{}
+
+var _ analysis.Analyzer = &Analyzer{} // TODO: Rename this with simply "Analyzer"
+
+var (
+	peerauthcollection    = collections.IstioSecurityV1Beta1Peerauthentications.Name() // TODO: Match the syntax name of the authpolicy collection, looks cleaner.
+	authpolicycollection  = collections.IstioSecurityV1Beta1Authorizationpolicies.Name()
+	requestauthcollection = collections.IstioSecurityV1Beta1Requestauthentications.Name()
+)
+
+func (a *Analyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "namespaceconflict.Analyzer",
+		Description: "Checks conditions related to Peer Authentication conflicting namespace level resources",
+		Inputs: collection.Names{
+			peerauthcollection,
+			authpolicycollection,
+			requestauthcollection,
+		},
+	}
+}
+
+func (a *Analyzer) Analyze(c analysis.Context) {
+	// Analyze collections.IstioSecurityV1Beta1Peerauthentications
+	c.ForEach(peerauthcollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.PeerAuthentication)
+
+		// If the resource has workloads associated with it, analyze for conflicts with selector
+		if x.GetSelector() != nil {
+			a.analyzeWorkloadSelectorConflicts(r, c)
+		}
+		return true
+	})
+
+	// Analyze collections.IstioSecurityV1Beta1Authorizationpolicies
+	c.ForEach(authpolicycollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.AuthorizationPolicy)
+
+		// If the resource has workloads associated with it, analyze for conflicts with selector
+		if x.GetSelector() != nil {
+			a.analyzeWorkloadSelectorConflicts(r, c)
+		}
+		return true
+	})
+
+	// Analyze collections.IstioSecurityV1Beta1Requestauthentications
+	c.ForEach(requestauthcollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.RequestAuthentication)
+
+		// If the resource has workloads associated with it, analyze for conflicts with selector
+		if x.GetSelector() != nil {
+			a.analyzeWorkloadSelectorConflicts(r, c)
+		}
+		return true
+	})
+}
+
+func (a *Analyzer) analyzeWorkloadSelectorConflicts(r *resource.Instance, c analysis.Context) {
+	// Find all resources that have the same selector
+	matches := a.findMatchingSelectors(r, c)
+
+	// There should be only one resource associated with a selector
+	if len(matches) != 0 {
+
+		// The namespace in which we will throw the conflict
+		xNS := r.Metadata.FullName.Namespace
+
+		// Cast the message to it's respective type to report the issue correctly
+		switch v := r.Message.(type) {
+		case *v1beta1.PeerAuthentication:
+			// Throw a Peer Authentication conflict
+			x := r.Message.(*v1beta1.PeerAuthentication)
+			m := msg.NewNamespaceResourceConflict(r, peerauthcollection.String(),
+				string(xNS), k8s_labels.SelectorFromSet(x.GetSelector().MatchLabels).String(),
+				matches.SortedList())
+			c.Report(collections.IstioSecurityV1Beta1Peerauthentications.Name(), m)
+			return
+		case *v1beta1.AuthorizationPolicy:
```

Comment:
I think it's actually allowed to have multiple authorization policies with the same selector? The semantics is additive and all authz policies will have effect.

Same as the RequestAuthentication policy.

Reply:
Definitely agree with you, but maybe we could show some best practices? I don't know I'm still pretty new here so I'd need some help figuring it out.

I think adding an Info or a Warning could guide the user in figuring out what potential things could go wrong.. Like spreading configs around multiple files? I don't know 😅.

What's your idea on that?

Reply:
Seems like someone could "forget" there was this other file... Eventhough it's not an error, it could be helpful in finding potential gaps in the configuration?